packages/hooks/src/Provider.tsx

```diff
@@ -0,0 +1,94 @@
+import { JsonRpcSigner } from '@ethersproject/providers/src.ts/json-rpc-provider';
+import React from 'react';
+import { ethers } from 'ethers';
+import WalletConnectProvider from '@walletconnect/web3-provider';
+import Web3Modal from 'web3modal';
+
+export interface Web3ContextType {
+  connectWallet: (() => void) | undefined;
+  signer: JsonRpcSigner | null | undefined;
+  userAddress: string | null | undefined;
+  disconnectWallet: (() => void) | undefined;
+  network: string;
+  chainId: number | undefined;
+  connected: boolean;
+  provider: ethers.providers.Web3Provider | null | undefined;
```

Comment:
this is a styling preference, so no need to accept, but i find it easier to grok these when the shorthand for undefined is used, like this:
```suggestion
  connectWallet?: (() => void);
  signer?: JsonRpcSigner | null;
  userAddress?: string | null;
  disconnectWallet?: (() => void);
  network: string;
  chainId?: number;
  connected: boolean;
  provider?: ethers.providers.Web3Provider | null;
```

Reply:
Yeah that def needs to be changed. Just laziness on my part. Wanted to ship this PR quick.